L1Trigger/L1TMuonEndCap/src/PrimitiveConversion.cc

```diff
@@ -679,48 +728,55 @@ void PrimitiveConversion::convert_gem(
     csc_nID += 1;
 
     if (tp_station == 1) {  // neighbor ME1
-      if (not(tp_subsector == 2))
-	{ edm::LogError("L1T") << "tp_subsector = " << tp_subsector; return; }
+      assert(tp_subsector == 2);
     }
   }
 
   // Set properties
-  conv_hit.SetGEMDetId     ( tp_detId );
-
-  conv_hit.set_endcap      ( (tp_endcap == 2) ? -1 : tp_endcap );
-  conv_hit.set_station     ( tp_station );
-  conv_hit.set_ring        ( tp_ring );
-  conv_hit.set_roll        ( tp_roll );
-  conv_hit.set_chamber     ( tp_chamber );
-  conv_hit.set_sector      ( tp_sector );
-  conv_hit.set_subsector   ( tp_subsector );
-  conv_hit.set_csc_ID      ( tp_csc_ID );
-  conv_hit.set_csc_nID     ( csc_nID );
-  //conv_hit.set_track_num   ( tp_data.trknmb );
-  //conv_hit.set_sync_err    ( tp_data.syncErr );
-
-  conv_hit.set_bx          ( tp_bx + bxShiftGEM_ );
-  conv_hit.set_subsystem   ( TriggerPrimitive::kGEM );
-  conv_hit.set_is_CSC      ( false );
-  conv_hit.set_is_RPC      ( false );
-  conv_hit.set_is_GEM      ( true  );
-
-  conv_hit.set_pc_sector   ( pc_sector );
-  conv_hit.set_pc_station  ( pc_station );
-  conv_hit.set_pc_chamber  ( pc_chamber );
-  conv_hit.set_pc_segment  ( pc_segment );
-
-  conv_hit.set_valid       ( true );
-  conv_hit.set_strip       ( tp_strip );
-  conv_hit.set_strip_low   ( tp_data.pad_low );
-  conv_hit.set_strip_hi    ( tp_data.pad_hi );
-  //conv_hit.set_wire        ( tp_data.keywire );
-  //conv_hit.set_quality     ( tp_data.quality );
-  conv_hit.set_pattern     ( 1 );  // In firmware, this marks GEM stub (unconfirmed!)
-  //conv_hit.set_bend        ( tp_data.bend );
-
-  conv_hit.set_neighbor    ( is_neighbor );
-  conv_hit.set_sector_idx  ( (endcap_ == 1) ? sector_ - 1 : sector_ + 5 );
+  //conv_hit.SetGEMDetId       ( tp_detId );
+  if (!is_me0) {
+    conv_hit.SetGEMDetId       ( tp_detId.getGEMDetId() );
+  } else {
+    conv_hit.SetME0DetId       ( tp_detId.getME0DetId() );
+  }
+
+  conv_hit.set_endcap        ( (tp_endcap == 2) ? -1 : tp_endcap );
+  conv_hit.set_station       ( tp_station );
+  conv_hit.set_ring          ( tp_ring );
+  conv_hit.set_roll          ( tp_roll );
+  conv_hit.set_chamber       ( tp_chamber );
+  conv_hit.set_sector        ( tp_sector );
+  conv_hit.set_subsector     ( tp_subsector );
+  conv_hit.set_csc_ID        ( tp_csc_ID );
+  conv_hit.set_csc_nID       ( csc_nID );
+  //conv_hit.set_track_num     ( tp_data.trknmb );
```

Comment:
Remove commented-out lines (also below)

Reply:
I keep these lines to list all the possible setters provided by the EMTFHit class. This makes it easier to remember to set it every time a new setter is added to the class. As you can just CTRL+F one of the existing setters, and just duplicate the line. I think it's good practice.

Reply:
Fair enought